public/frontend/script.js

```diff
@@ -0,0 +1,330 @@
+let chatrooms = [];
+let allUsers = [];
+let currentlyOpenChatroom = {};
+let newMessageContent = "";
+
+const activeUser = localStorage.getItem("currentUser"); // handle with care
+const token = localStorage.getItem("token");
+
+const socket = io("http://localhost:5500/chats", { auth: { token } });
```

Comment:
Why is the host hardcoded? How will you learn this in production?

Reply:
unfortunately we can't use environment variable here because its on the frontend page i created for the demo and i cant access the .env file file from this point but again i  forgot to switch it to the deployed link but i just did fix it.Thanks!!